controllers/v1/observationsController.js

```diff
@@ -631,128 +630,140 @@ module.exports = class Observations extends Abstract {
                 let messageData = messageConstants.apiResponses.ENTITY_FETCHED;
 
                 if( !(userAllowedEntities.length > 0) && req.query.parentEntityId ) {
-
+                    
                     let entityType = entitiesHelper.entitiesSchemaData().SCHEMA_ENTITY_GROUP+"."+result.entityType;
+                    
+                    let filterData = {
+                        "id" : req.query.parentEntityId
+                    };
 
-                    let entitiesData = await entitiesHelper.entityDocuments({
-                        _id:req.query.parentEntityId
-                      }, [
-                        entityType,
-                        "entityType",
-                        "metaInformation.name",
-                        "metaInformation.addressLine1",
-                        "metaInformation.addressLine2",
-                        "metaInformation.externalId",
-                        "metaInformation.districtName"
-                      ]);
-
-                    if( entitiesData.length > 0 && entitiesData[0].groups && entitiesData[0].groups[result.entityType]  ) {
-                        userAllowedEntities = 
-                        entitiesData[0][entitiesHelper.entitiesSchemaData().SCHEMA_ENTITY_GROUP][result.entityType];
-                    } else {
+                    let entitiesDocument = await sunbirdService.learnerLocationSearch( filterData );
 
+                    if ( !entitiesDocument.data.response > 0 ) {
```

Comment:
@VISHNUDAS-tunerlabs check the timeout error, check data present or not, check based on success key